server/src/collection.rs

```diff
@@ -33,3 +52,31 @@ impl Collection {
         todo!()
     }
 }
+
+impl Collection {
+    pub fn new(label: String, alias: String, manager: Arc<Mutex<ServiceManager>>) -> Self {
+        let created = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap();
+
+        Self {
+            items: Default::default(),
+            label: Mutex::new(label.clone()),
+            locked: AtomicBool::new(true),
+            modified: created,
+            alias: Mutex::new(alias),
+            n_items: RwLock::new(0),
```

Comment:
Like n_sesssions, you have the list of the items. You can get the n_items...

Reply:
`/org/freedesktop/secrets/collection/login/iiii`

Yeah. The only thing that I don't like about this approach (counting the items to set the last part of an objectpath) is `iiii` should be unique/specific. With this approach there is always a chance (when an item is deleted and then another created, the object tree will be the same) that two items has the same `iiii` and so the the same objectpath. 

Reference: https://specifications.freedesktop.org/secret-service-spec/latest/object-paths.html

Reply:
Then this should at least use Default::default then.

Reply:
Well, the spec says auto-generated not necessarily the counter. So maybe we should use some uuid or random string.

Reply:
Yeah, but at least for now, I'd like to follow the gnome-keyring approach. It's simple and we can have a clearer object tree with this approach :)